tb_device_http.py

```diff
@@ -94,6 +115,137 @@ def log_level(self, value: typing.Union[int, str]):
         self.logger.setLevel(value)
         self.logger.critical('Log level set to %s', self.log_level)
 
+    def get_firmware_info(self):
+        response = self.__session.get(
+            f"{self.__config['host']}/api/v1/{self.__config['token']}/attributes",
+            params={"sharedKeys": REQUIRED_SHARED_KEYS}).json()
+        return response.get("shared", {})
+
+    def get_firmware(self, fw_info):
+        chunk_count = ceil(fw_info.get(FW_SIZE_ATTR, 0) / self.chunk_size) if self.chunk_size > 0 else 0
+        firmware_data = b''
+        for chunk_number in range(chunk_count + 1):
+            params = {"title": fw_info.get(FW_TITLE_ATTR),
+                      "version": fw_info.get(FW_VERSION_ATTR),
+                      "size": self.chunk_size if self.chunk_size < fw_info.get(FW_SIZE_ATTR,
+                                                                               0) else fw_info.get(
+                          FW_SIZE_ATTR, 0),
+                      "chunk": chunk_number
+                      }
+            self.logger.debug(params)
+            self.logger.debug(
+                'Getting chunk with number: %s. Chunk size is : %r byte(s).' % (chunk_number + 1, self.chunk_size))
+            response = self.__session.get(
+                f"{self.__config['host']}/api/v1/{self.__config['token']}/firmware",
+                params=params)
+            if response.status_code != 200:
+                self.logger.error('Received error:')
+                response.raise_for_status()
+                return
+            firmware_data = firmware_data + response.content
+        return firmware_data
+
+    def on_firmware_received(self, firmware_info, firmware_data):
+        with open(firmware_info.get(FW_TITLE_ATTR), "wb") as firmware_file:
+            firmware_file.write(firmware_data)
+
+        self.logger.info('Firmware is updated!\n Current firmware version is: %s' % firmware_info.get(FW_VERSION_ATTR))
+
+    def verify_checksum(self, firmware_data, checksum_alg, checksum):
```

Comment:
It looks like this code duplicates with file tb_device_mqtt.py. Could we move duplicated code to some extra file, e.g. sdk_utils.py?